lib/auth/access.go

```diff
@@ -77,8 +85,45 @@ func (a *Server) DeleteRole(ctx context.Context, name string) error {
 				// Mask the actual error here as it could be used to enumerate users
 				// within the system.
 				log.Warnf("Failed to delete role: role %v is used by user cert authority %v", name, a.GetClusterName())
-				return trace.BadParameter("failed to delete a role that is still in use by a user, check the system server logs for more details")
+				return trace.Wrap(errDeleteRoleCA)
+			}
+		}
+	}
+
+	var nextToken string
+	for {
```

Comment:
Is there no "forEach" kind of helper method for access lists already somewhere?

Reply:
No, unfortunately. :-(